src/Microsoft.PowerShell.SHiPS/SHiPSProvider.cs

```diff
@@ -209,8 +231,226 @@ protected override IEnumerable<IPathNode> GetNodeFactoryFromPathForItemExists(st
 
             var context = CreateContext(path);
             return shipsPathResolver.ResolvePath(context, path, context.Force);
+        }
+
+        #endregion
+
+        #region NotSupportedCommands
+
+        // Implement IContentCmdletProvider. 
+        // As these methods are not virtual in base class and we don't want to change base class, I use New and call the interface to enforce the polymorphism.
+        // More Explain: https://stackoverflow.com/questions/12314990/override-method-implementation-declared-in-an-interface
+        #region IContentCmdletProvider - Get/Set/Clear-Content
+
+        public new void ClearContent(string path)
+        {
+            var errorMsg = GetNotSupportedMessage("Clear-Content", path);
+            WriteErrorNotImplemented(errorMsg);
+        }
+
+        public new object ClearContentDynamicParameters(string path)
+        {
+            ClearContent(path);
+            return null;
+        }
+
+        void IContentCmdletProvider.ClearContent(string path)
+        {
+            ClearContent(path);

```

Comment:
Curiousity:
Would `this.ClearContent(path)` increase readability?

If so, the comment applies all over. If not, then none need to change.